File: src/main/java/com/example/marketplace/services/cart/CartService.java
package com.example.marketplace.services.cart;

import com.example.marketplace.models.*;

public interface CartService {
    Cart getCartByUserId(Long userId);
    Cart addProductToCart(CartProduct cartProduct, User user);
    Cart removeProductFromCart(Long id, Long userId);
    Cart updateProductQuantityInCart(Long userId, Long cartProductId, int newQuantity);
    Cart saveAddressToCart(Long userId, Address address);
    Cart savePaymentToCart(Long userId, Payment payment);
    void deleteCart(Long userId);
    int getCartProductLengthByUserId(Long userId);
}

File: frontend/src/services/cartService.js
import httpClient from "./httpClient";
import authHeader from "./authHeader";

export default {
  getCart() {
    return httpClient.get("/cart", { headers: authHeader() });
  },
  addProductToCart(body) {
    return httpClient.post("/cart/cart-product", body, {
      headers: authHeader(),
    });
  },
  removeProductFromCart(id) {
    const uri = `/cart/cart-product/${id}`;
    return httpClient.delete(uri, { headers: authHeader() });
  },
  deleteCart() {
    return httpClient.delete("/cart", { headers: authHeader() });
  },
  addAddressToCart(body) {
    return httpClient.put("/cart/address", body, { headers: authHeader() });
  },
  addPaymentToCart(body) {
    return httpClient.put("/cart/payment", body, { headers: authHeader() });
  },
  updateProductQuantityInCart(id, newQuantity) {
    const uri = `/cart/cart-product/${id}`;
    return httpClient.put(uri, { newQuantity }, { headers: authHeader() });
  },
  getCartProductLengthByUserId() {
    return httpClient.get("/cart/length", { headers: authHeader() });
  },
};
